src/main/scala/com/codacy/api/service/CoverageServices.scala

```diff
@@ -13,17 +13,52 @@ class CoverageServices(client: CodacyClient) {
       partial: Boolean = false
   ): RequestResponse[RequestSuccess] = {
     val endpoint = s"coverage/$commitUuid/${language.toLowerCase}"
-    val queryParams = Map("partial" -> partial.toString)
 
-    val jsonString = Json.stringify(Json.toJson(coverageReport))
-
-    client.post(Request(endpoint, classOf[RequestSuccess], queryParams), jsonString)
+    PostRequest(endpoint, coverageReport, partial)
   }
 
   def sendFinalNotification(commitUuid: String): RequestResponse[RequestSuccess] = {
     val endpoint = s"commit/$commitUuid/coverageFinal"
 
-    client.post(Request(endpoint, classOf[RequestSuccess]), "{}")
+    PostEmptyRequest(endpoint)
+  }
+
+  def sendReportWithProjectName(
```

Comment:
Since we don't have a way of specifying the required token (account vs project). It would be nice to have documentation on the methods declaring which token the endpoints require

Reply:
Done